perllib/FixMyStreet/Roles/CobrandEcho.pm

```diff
@@ -609,6 +609,57 @@ sub clear_cached_lookups_bulky_slots {
     $echo->CancelReservedSlotsForEvent($guid);
 }
 
+sub bulky_refetch_slots {
+    my ($self, $row, $verbose) = @_;
+
+    my $property_id = $row->get_extra_field_value('property_id');
+    my $date = $self->collection_date($row);
+    my $guid = $row->get_extra_field_value('GUID');
+    my $window = $self->_bulky_collection_window();
+
+    my $cfg = $self->feature('echo');
+    my $echo = Integrations::Echo->new(%$cfg);
+
+    my $service_id = $cfg->{bulky_service_id};
+    my $event_type_id = $cfg->{bulky_event_type_id};
+
+    if ($row->whensent || !$guid) {
+        if ($row->whensent) {
+            say "Already sent, creating new GUID and fetching new reservation and resending" if $verbose;
+        } else {
+            say "No GUID? Creating new GUID and fetching new reservation and sending" if $verbose;
+        }
+        require UUID::Tiny;
+        $guid = UUID::Tiny::create_uuid_as_string();
+        $row->update_extra_field({ name => 'GUID', value => $guid });
+        $row->state('confirmed');
+        $row->resend;
+    } else {
+        say "Not already sent, fetching new reservation and trying again" if $verbose;
+        $p->send_fail_count(0); # Assuming it's been failing, for an instant retry
```

Comment:
Should be `$row` I think.

```suggestion
        $row->send_fail_count(0); # Assuming it's been failing, for an instant retry
```

Reply:
Whoops, copied from live script. Also moved it further down to only do this if slot found to prevent the repetition you raise.